### DISCUSSION TITLE
Diagnostics

### DISCUSSION THREAD

**Discussion by tiborauer** [automaticanalysis/automaticanalysis#223] [Ideas]
I have noticed that there is an inconsistency in placing diagnostics, which is bad, because some of them are ignored during the report generation. Some of them are in the subjects' folders, some of them are in the module's folder, and some of them are in a separate "diagnistics" folder. I personally prefer:
1. subjetc-level diagnostics going to the subjetcs' folders,
2. mudule-level summaries/statistics going to the corresponding module's folder
to keep results and diagnostics in the same place.

Another option could be using "diagnostics" folder both
1. in the subjetcs' folders (for the subjetc-level diags.) and
2. in the corresponding module's folder (for the summaries),
and puttign the diagnostics there.

**Comment by alexvicegrab** [comment]
The diagnostic folder in the module folder is my idea/fault. I thought it would be useful to have all subjects'/sessions' diagnostics available and located in one single place for easy viewing (before any diagnostic report thing was in working state)

**Comment by tiborauer** [comment]
I think we should settle this issue, because I planning to add diagnostics for ruther modules and it would be good to have a consistency.
Does anyone else has an opinion/vote on this? 
My further argument for keeping diagnostics and results at the same place is so that it is easier to locate them for report generation.

**Comment by jpeelle** [comment]
Bumping this up for general discussion. We discussed:

- Being consistent across all modules about where we save diagnostic images, and how they are named
- Possibly expanding reporting to have different levels of granularity or detail, either as option in headers/aap, or by having a slightly more complicated html structure with multiple pages

The main thing is to be consistent and continue to improve the reporting (which we find very useful!).
